src/main/java/hudson/plugins/logparser/LogParserPublisher.java

```diff
@@ -61,64 +68,95 @@ private LogParserPublisher(final boolean unstableOnWarning,
         this.projectRulePath = projectRulePath;
     }
 
+    @DataBoundConstructor
+    public LogParserPublisher(boolean useProjectRule, String projectRulePath, String parsingRulesPath) {
+        super();
+        if (useProjectRule) {
+            this.projectRulePath = Util.fixEmpty(projectRulePath);
+            this.parsingRulesPath = null;
+        } else {
+            this.parsingRulesPath = Util.fixEmpty(parsingRulesPath);
+            this.projectRulePath = null;
+        }
+        this.useProjectRule = useProjectRule;

```

Comment:
Hmm, switch blocks in a data-bound constructor is an antipattern. Nicer would be to either

* Have one `rulePath` field, with a flag controlling how it is used. (flat data style)
* Or, have a nested `abstract` `Describable` called something like `Rule`, with two concrete implementations, each with their own `config.jelly`. (OO style)